File: src/Entreprise.php
<?php

namespace Carfleet;

class Entreprise
{
    #region private attributes
    private array $vehicules = [];
    private array $drivers = [];
    #endregion private attributes

    private function getVehiculeByChassisNumber(string $chassisNumber): Vehicule
    {
        foreach ($this->vehicules as $vehicule) {
            if ($vehicule->chassisNumber === $chassisNumber) {
                return $vehicule;
            }
        }
        throw new VehicleNotFoundException();
    }

    private function getDriverByEmailAddress(string $driverEmailAddress): Driver
    {
        foreach ($this->drivers as $driver) {
            if ($driver->emailAddress === $driverEmailAddress) {
                return $driver;
            }
        }
        throw new DriverNotFoundException();
    }

    public function assignVehiculeToDriver(string $chassisNumber, string $driverEmailAddress): void
    {
        $driver = $this->getDriverByEmailAddress($driverEmailAddress);
        $vehicule = $this->getVehiculeByChassisNumber($chassisNumber);
        
        // Logic to assign the vehicle to the driver goes here
    }

    #region nested classes
    class EnterpriseException extends Exception {}

    class VehicleNotFoundException extends EnterpriseException {}

    class DriverNotFoundException extends EnterpriseException {}
    #endregion nested classes
}


/*
class Vehicule
{
    public string $chassisNumber;

    public function __construct(string $chassisNumber)
    {
        $this->chassisNumber = $chassisNumber;
    }
}

class Driver
{
    public string $emailAddress;

    public function __construct(string $emailAddress)
    {
        $this->emailAddress = $emailAddress;
    }
} décalé dans les autres classes*/

File: src/Entreprise.cs
namespace Carfleet
{
    public class Entreprise
    {
        #region private attributes
        private List<Vehicule> _vehicule = new List<Vehicule>()
        private List<Driver> _driver = new List<Driver>()
        private List<Truck> _truck = new List<Truck>()
        private List<TruckDriver> _truckdriver = new List<TruckDriver>()
        #endregion private attributes
        private Vehicule VehiculeByChassisNumber(string chassisNumber)
        {
            foreach (Truck truck in _truck)
            {
                if (truck.chassisNumber == chassisNumber)
                {
                    return truck;
                }
            }
            throw new VehicleNotFoundException();
        }

        private Driver GetDriverByEmailaddress(string driverEmailAddress)
        {
            foreach (Truckdriver truckdriver in _truckdriver)
            {
                if (truckdriver.emailaddress == driverEmailAddress)
                {
                    return truckdriver;
                }
            }
            throw new DriverNotFoundException();
        }
        public string getDriverByEmailaddress { get => emailaddress}


        #region public methods

        public function assignVehiculeToDriver(string chassisNumber, string driverEmailAddress) : void
        {
            Driver driver = this.getDriverByEmailaddress(driverEmailAddress);
    }


    #endregion public methods

    #region private methods
    #endregion private methods

    #region nesteded class
    public class EnterpriseException : Exception { }
    public class VehicleNotFoundException : EnterpriseException { }
    public class DriverNotFoundException : EnterpriseException { }
    #endregion nesteded class
}
}
